pkg/mock/phase_pending_test.go

```diff
@@ -0,0 +1,297 @@
+package mock
+
+import (
+	"testing"
+	"time"
+
+	v1 "github.com/atlassian-labs/cyclops/pkg/apis/atlassian/v1"
+	"github.com/stretchr/testify/assert"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Basic test to ensure the base functionality of the Pending phase works. A
+// predictable set of nodes with matching cloud provider instances attached to
+// their nodegroups should get the CNR transitioned to the Initialized phase.
+func TestPendingSimpleCase(t *testing.T) {
+	nodegroup, err := newNodegroup("ng-1", 2)
+	if err != nil {
+		assert.NoError(t, err)
+	}
+
+	mockTransitioner := NewMockTransitioner(
+		WithCycleNodeRequest(&v1.CycleNodeRequest{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cnr-1",
+				Namespace: "kube-system",
+			},
+			Spec: v1.CycleNodeRequestSpec{
+				NodeGroupsList: []string{"ng-1"},
+				CycleSettings: v1.CycleSettings{
+					Concurrency: 1,
+					Method:      v1.CycleNodeRequestMethodDrain,
+				},
+				Selector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"customer": "kitt",
+					},
+				},
+			},
+		}),
+		StartAtPhase(v1.CycleNodeRequestPending),
+		WithKubeNodes(nodegroup),
+		WithCloudProviderInstances(nodegroup),
+	)
+
+	result, err := mockTransitioner.Run()
+	assert.NoError(t, err)
+	assert.True(t, result.Requeue)
+
+	// It should move to the Initialised phase and set up the status of the CNR
+	// in a predictable manner
+	assert.Equal(t, v1.CycleNodeRequestInitialised, mockTransitioner.Cnr.Status.Phase)
+	assert.Len(t, mockTransitioner.Cnr.Status.NodesToTerminate, 2)
+	assert.Equal(t, mockTransitioner.Cnr.Status.ActiveChildren, int64(0))
+	assert.Equal(t, mockTransitioner.Cnr.Status.NumNodesCycled, 0)
+}
+
+// Test to ensure the Pending phase will reject a CNR with a named node that
+// does not match any of the nodes matching the node selector. It should error
+// out immediately.
+func TestPendingWrongNamedNode(t *testing.T) {
+	nodegroup, err := newNodegroup("ng-1", 2)
+	if err != nil {
+		assert.NoError(t, err)
+	}
+
+	mockTransitioner := NewMockTransitioner(
+		WithCycleNodeRequest(&v1.CycleNodeRequest{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cnr-1",
+				Namespace: "kube-system",
+			},
+			Spec: v1.CycleNodeRequestSpec{
+				NodeGroupsList: []string{"ng-1"},
+				CycleSettings: v1.CycleSettings{
+					Concurrency: 1,
+					Method:      v1.CycleNodeRequestMethodDrain,
+				},
+				Selector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"customer": "kitt",
+					},
+				},
+				NodeNames: []string{
+					"node-0",
```

Comment:
I guess one nit here is in this test we're assuming that when you create a test node group with `newNodegroup()` that nodes follow this naming scheme. 

Someone modifying `newNodegroup()` may find they break a ton of tests if they decide to change an assumption on how test node groups are created

Reply:
You could possibly remedy this by adding a node that obviously would very obviously never exist in the `NodeNames` field